### DISCUSSION TITLE
Disable `audio_cache`

### DISCUSSION THREAD

**Discussion by nkilm** [speechbrain/speechbrain#2570] [General]
Is there an option to disable caching? Also I don't see any option to edit the path where the audio files are to be cached. I'm running [Speech Enhancement Pipeline](https://huggingface.co/speechbrain/sepformer-whamr-enhancement). 

```python
# https://huggingface.co/speechbrain/sepformer-whamr-enhancement

from speechbrain.inference.separation import SepformerSeparation as separator
import torchaudio

model = separator.from_hparams(source="path-to-model", savedir='path-to-model/pretrained_models/')

# for custom file, change path
# this internally calls fetch() for given input path and caches in `audio_cache`
est_sources = model.separate_file(path='input-file.wav') 

torchaudio.save("enhanced_whamr.wav", est_sources[:, :, 0].detach().cpu(), 8000)
```

The below function is called which creates a sym-link in `audio_cache` folder which points to original source. 
https://github.com/speechbrain/speechbrain/blob/1bd75e4b5e10e0d45bcf06f9b9b4564341530dbe/speechbrain/utils/fetching.py#L54-L63

<br/>

The problem I'm facing is I'm testing this pipeline under different Linux user who does not have the _write_ permissions to the directory where the code resides. The _write_ permissions are required in order to create the sym-link or cache the input audio file.

**Comment by asumagic** [comment]
This is linked to https://github.com/speechbrain/speechbrain/pull/2476 which is still a work in progress (should make it into v1.0.2 in a while)

Currently, `load_audio` in inference classes create a symlink.

You could work around this by looking into the code for `separate_file` and replicate the code in a way that just loads the audio directly without calling into `fetch()`.

Alternatively, the linked PR should already avoid the issue for your usecase, as it would avoid creating a symlink entirely for a local file.  
You could make a local SB install that points to this branch. Something like:

```bash
git clone https://github.com/asumagic/speechbrain.git -b symlinkinator
pip install --editable speechbrain/
```

**Reply by nkilm** [reply]
Great, I'll look into this!

Thank you @asumagic!
